grudge/dt_utils.py

```diff
@@ -45,22 +57,65 @@
 from pytools import memoize_on_first_arg
 
 
+def characteristic_lengthscales(dcoll: DiscretizationCollection) -> DOFArray:
```

Comment:
I feel like this would be a candidate for memoization, moreso than `dt_geometric_factors`.